File: 41-50/Q43.cpp
class Solution {
public:
    string multiply(string a, string b) {
        if(a=="0" || b=="0") return "0";
        int n=a.size(), m=b.size();
        vector<int> res(n+m, 0);

        for(int i=n-1;i>=0;i--){
            int x = a[i]-'0';
            for(int j=m-1;j>=0;j--){
                int y = b[j]-'0';
                int sum = res[i+j+1] + x*y;
                res[i+j+1] = sum % 10;
                res[i+j] += sum / 10;
            }
        }

        string s;
        int i=0;
        while(i<res.size() && res[i]==0) i++;
        for(; i<res.size(); i++) s.push_back(res[i]+'0');
        return s;
    }
};

File: 41-50/Q43.py
#Q43.py
class Solution:
    def multiply(self, a: str, b: str) -> str:
        if a == "0" or b == "0":
            return "0"
        n, m = len(a), len(b)
        res = [0]*(n+m)

        for i in range(n-1, -1, -1):
            x = ord(a[i]) - 48
            for j in range(m-1, -1, -1):
                y = ord(b[j]) - 48
                s = res[i+j+1] + x*y
                res[i+j+1] = s % 10
                res[i+j] += s // 10

        i = 0
        while i < len(res) and res[i] == 0:
            i += 1
        return "".join(str(d) for d in res[i:])
